.github/workflows/integration.yml

```diff
@@ -5,12 +5,39 @@ on:
     branches:
       - 'master'
       - '[0-9].[0-9]+'  # release branches
+      - '**-full-ci'
     tags:
-      - '*'
+      - '**'
+  pull_request:
+    types: [ opened, reopened, synchronize, labeled ]
   workflow_dispatch:
 
+concurrency:
+  # Update of a developer branch cancels the previously scheduled workflow
+  # run for this branch. However, the 'master' branch, release branch (1.10,
+  # 2.8, etc.), and tag workflow runs are never canceled.
+  #
+  # We use a trick here: define the concurrency group as 'workflow run ID' +
+  # 'workflow run attempt' because it is a unique combination for any run.
+  # So it effectively discards grouping.
+  #
+  # Important: we cannot use `github.sha` as a unique identifier because
+  # pushing a tag may cancel a run that works on a branch push event.
+  group: ${{ (
+    github.ref == 'refs/heads/master' ||
+    github.ref == 'refs/heads/1.10' ||
+    startsWith(github.ref, 'refs/heads/2.') ||
+    startsWith(github.ref, 'refs/tags/')) &&
+    format('{0}-{1}', github.run_id, github.run_attempt) ||
+    format('{0}-{1}', github.workflow, github.ref) }}
+  cancel-in-progress: true
+
 jobs:
   tarantool:
+    # Run on pull request only if the 'full-ci' label is set.
+    if: github.event_name != 'pull_request' ||
+      contains(github.event.pull_request.labels.*.name, 'full-ci')
+
```

Comment:
What about other jobs (vshard, cartridge, etc)? Shouldn't we run them as well?

Reply:
Oh, I see that they do run now. Sorry for the noise. The patch LGTM.

Reply:
They will not run because they depend on the 'tarantool' job. If the 'tarantool' job is skipped, any job that needs it will be skipped as well.